dataprovider/src/main/java/cm/aptoide/pt/dataprovider/ws/v7/QueryStringMapper.java

```diff
@@ -0,0 +1,92 @@
+package cm.aptoide.pt.dataprovider.ws.v7;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryStringMapper {
+
```

Comment:
Not really important, but the way this mapper is being used, there's no advantage to instantiating it. I'd either just make it static (since really they're basically stateless functions, and you're calling `new QueryStringMapper()` everywhere), or maybe instantiate it only one time on the V7 class (since it seems to be universal throughout requests) and retrieve it from there when necessary.

Reply:
I forgot to extract it :) will do it

Reply:
I ended up caching the QueryStringMapper in v7.java and using it everywhere. I wanted to pull it to dagger but that would potentially involve doing a major refactor in all the requests.